### PR TITLE
Make colorbar compatible accepting a list of axes created via gridspec

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Similar to #8739, but modifies `make_axes_gridspec` in `colorbar.py`.

This is a change to make `figure.colorbar(ppm,ax=parents)` accept a list of parents that are `SubplotBase`, and were created by `gridspec` and return a colorbar that is `SubplotBase`.  So a call like:

```python
fig, axs = plt.subplots(2,2)
pcm = axs[0,0].pcolormesh(np.random(32,32))
fig.colorbar(ppm, ax=axs)
fig.tight_layout()
```
works with `axs` as a list, and `colorbar` taking space from all the axes listed by `axs`, and `tight_layout()` able to operate on all axes including the colorbar. 

I think it works for most edge cases.  I hesitated to edit the test suite, but could give that a try if desired.  

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
Can you squash this to one or two commits?

File: lib/matplotlib/colorbar.py [hunk 1]
```diff
@@ -1165,24 +1165,36 @@ def make_axes(parents, location=None, orientation=None, fraction=0.15,
...
+
+
+def rowcolunm2index(row, col, ncols):
+    index = col-1+(row-1)*ncols
+    return index
+
+
 @docstring.Substitution(make_axes_kw_doc)
-def make_axes_gridspec(parent, **kw):
+def make_axes_gridspec(parents, **kw):
```
Comment (tacaswell) [inline lib/matplotlib/colorbar.py]: Changing the argument name is an API change (as `make_akes_gridspec(parent=obj)` will be broken).

File: lib/matplotlib/colorbar.py [hunk 2]
```diff
@@ -1213,17 +1225,52 @@ def make_axes_gridspec(parent, **kw):
     pad_s = (1. - shrink) * 0.5
     wh_ratios = [pad_s, shrink, pad_s]
 
+    # make parents a 1-d ndarray if its not already...
+    parents = np.atleast_1d(parents).ravel()
+    # For the subplotspec that these axes belong to, loop through and get
+    # the maximum and minimum index into the subplotspec.  The result is the
+    # size the new gridspec that we will create must be.
+    gsp0 = parents[0].get_subplotspec().get_gridspec()
+    minind = 10000
```
Comment (tacaswell) [inline lib/matplotlib/colorbar.py]: Probably better to use `np.inf` and `-np.inf` or use the values from the first parent.

File: setup.cfg.template [hunk 3]
```diff
@@ -13,19 +13,19 @@
 # set this to True.  It will download and build a specific version of
 # FreeType, and then use that to build the ft2font extension.  This
 # ensures that test images are exactly reproducible.
-#local_freetype = False
+local_freetype = True
```
Comment (tacaswell) [inline setup.cfg.template]: These changes should not be committed

**Comment by jklymak** [comment]
Hmm, apparently not..  added a bunch of other files somehow.  Sigh

**Comment by tacaswell** [comment]
You need to force-push to github, at some point when you are trying to push the rebased branch it is saying "Can not push, please merge upstream changes first"?

See https://github.com/matplotlib/matplotlib/pull/2742 for detailed instructions on how to deal with rebasing (which has annoyingly been removed from our built documentation...)

**Comment by jklymak** [comment]
yeah, but I did my rebase wrong.  I foolishly thought I didn't have to count the commits...  I'm going to start over.  Sorry for my bad git skills

**Comment by tacaswell** [comment]
I strongly recommend using `gitk` or other GUI tool that will show you the actual graph (`git log --graph`) rather than the flattened version github shows you.

Cherry-picking ranges can also be superhelpful for fixing this sort of thing.

http://tom.preston-werner.com/2009/05/19/the-git-parable.html <- this is what made me actually understand git.

**Comment by tacaswell** [comment]
and making git-snarls is an unfortunate, but real part of learning to use git.

**Comment by jklymak** [comment]
I'll maintain that I understand git just fine, but git doesn't understand me ;-)

**Pull request closed without merging**